lmp/fetch_root_meta.py

```diff
@@ -0,0 +1,75 @@
+#!/usr/bin/python3
+#
+# Copyright (c) 2022 Foundries.io
+# SPDX-License-Identifier: Apache-2.0
+
+import sys
+import os
+import logging
+import traceback
+import argparse
+import requests
+
+logger = logging.getLogger(__name__)
+
+
+def get_params():
+    parser = argparse.ArgumentParser('Fetch TUF root role metadata')
+    parser.add_argument('-f', '--factory', help='Factory name', required=True)
+    parser.add_argument('-a', '--token-file', help='A file containing Factory API Token, aka OSF Token', required=True)
+    parser.add_argument('-d', '--dst-dir', help='Directory to download metadata to', required=True)
+
+    return parser.parse_args()
+
+
+def fetch_all_root_meta(factory: str, token: str, dst: str, prod: bool = False):
+    api_base_url = f'https://api.foundries.io/ota/repo/{factory}/api/v1/user_repo/'
+    ver = 1
+    while True:
+        meta_file = f"{ver}.root.json"
+        logger.info(f"Fetching {meta_file}")
+        response = requests.get(os.path.join(api_base_url, meta_file),
+                                params={'production': 1} if prod else None, headers={'osf-token': token})
+        if response.status_code == 404:
+            break
+
+        if not response.ok:
+            raise requests.exceptions.HTTPError('Failed to get root meta {}: HTTP_{}\n{}'.
+                                                format(meta_file, response.status_code, response.text))
+
+        with open(os.path.join(dst, meta_file), "wb") as f:
+            f.write(response.content)
+        ver += 1
+
+
+def main():
+    exit_code = os.EX_OK
+    try:
+        params = get_params()
+        logger.info(f'Starting fetching TUF root role metadata; factory: {params.factory}, dst-dir: {params.dst_dir}')
+
+        with open(params.token_file) as f:
+            token = f.read().strip()
+
+        for meta_type in [ ("ci", False), ("prod", True)]:
+            dst = os.path.join(params.dst_dir, meta_type[0])
+            logger.info(f"Fetching {meta_type[0]} root metadata to {dst}")
+            os.makedirs(dst, exist_ok=True)
+            fetch_all_root_meta(params.factory, token, dst, meta_type[1])
+
+        logger.info(f'TUF root role metadata has been successfully downloaded; factory: {params.factory}, dst-dir: {params.dst_dir}')
+    except Exception as exc:
+        logger.error('Failed to fetch TUF root role metadata: {}\n{}'.format(exc, traceback.format_exc()))
+        exit_code = os.EX_SOFTWARE
+
+    return exit_code
+
+
+if __name__ == '__main__':
+    log_formatter = logging.Formatter('%(asctime)s %(levelname)s: %(module)s: %(message)s')
+    stream_hdl = logging.StreamHandler(sys.stdout)
+    stream_hdl.setFormatter(log_formatter)
+    logging.root.addHandler(stream_hdl)
+    logging.root.setLevel('INFO')
```

Comment:
Then logging from other modules won't work, if any.
I will have to change this part anyway, since I need to log into a file since bitbake doesn't output `stdout`.